resources/assets/coffee/react/beatmapset-page/scoreboard-table.coffee

```diff
@@ -16,11 +16,37 @@
 #    along with osu!web.  If not, see <http://www.gnu.org/licenses/>.
 ###
 
-{a, div, table, tr, td, th, thead, tbody} = ReactDOMFactories
+{a, button, i, span, table, tr, td, th, thead, tbody} = ReactDOMFactories
 el = React.createElement
 bn = 'beatmap-scoreboard-table'
 
 BeatmapsetPage.ScoreboardTable = (props) ->
+  props.scores = props.scores.map (score) ->
+    score.reportButton =
+      if score.userReportPresence.reported
```

Comment:
userReportPresence will be undefined if not logged in.

Fetching the presence information is doing an an extra query per score adding up to a lot of extra queries.